pymesos/process.py

```diff
@@ -202,6 +202,11 @@ def on_event(self, event):
     def on_close(self):
         raise NotImplementedError
 
+    def process_event(self, event):
+        with self._lock:
+            if not self._stop:
+                self.on_event(event)
```

Comment:
maybe we should. first of all, callbacks and IO are now in the same thread, so most of the time we do not have races here. this lock is just to prevent we call `stop` in other thread when we are doing callback in this scenario, synchronization is needed, I think.